tests/arp/test_arp_dualtor.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import ptf.testutils as testutils
+import pytest
+import time
+
+from ipaddress import ip_network, IPv6Network, IPv4Network
+from tests.arp.arp_utils import clear_dut_arp_cache, increment_ipv6_addr, increment_ipv4_addr
+from tests.common.helpers.assertions import pytest_assert
+
+pytestmark = [
+    pytest.mark.topology(('t0', 'dualtor'))
+]
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def setup_ptf_proxy_arp(config_facts, ptfhost, intfs_for_test):
+    _, intf1_indice, _, _, _, _, _ = intfs_for_test
+    ip_addr_config_cmd = 'ip addr {} {}/{} dev {}'
+
+    # Calculate the IPv6 address to assign to the PTF port
+    vlan_addrs = config_facts['VLAN_INTERFACE'].items()[0][1].keys()
+    intf_ipv6_addr = None
+    intf_ipv4_addr = None
+
+    for addr in vlan_addrs:
+        if type(ip_network(addr, strict=False)) is IPv6Network:
+            intf_ipv6_addr = ip_network(addr, strict=False)
+        elif type(ip_network(addr, strict=False)) is IPv4Network:
+            intf_ipv4_addr = ip_network(addr, strict=False)
+
+    # The VLAN interface on the DUT has an x.x.x.1 address assigned (or x::1 in the case of IPv6)
+    # But the network_address property returns an x.x.x.0 address (or x::0 for IPv6) so we increment by two to avoid conflict
+    ptf_intf_ipv4_addr = increment_ipv4_addr(intf_ipv4_addr.network_address, incr=2)
+    ptf_intf_ipv6_addr = increment_ipv6_addr(intf_ipv6_addr.network_address, incr=2)
+    ptf_intf_name = "eth{}".format(intf1_indice)
+
+    logger.info("Configuring {} and {} on PTF interface {}".format(ptf_intf_ipv4_addr, ptf_intf_ipv6_addr, ptf_intf_name))
+    ptfhost.shell(ip_addr_config_cmd.format('add', ptf_intf_ipv4_addr, intf_ipv4_addr.prefixlen, ptf_intf_name))
+    ptfhost.shell(ip_addr_config_cmd.format('add', ptf_intf_ipv6_addr, intf_ipv6_addr.prefixlen, ptf_intf_name))
+
+    yield ptf_intf_ipv4_addr, ptf_intf_ipv6_addr, ptf_intf_name 
+
+    logger.info("Removing {} and {} from PTF interface {}".format(ptf_intf_ipv4_addr, ptf_intf_ipv6_addr, ptf_intf_name))
+    ptfhost.shell(ip_addr_config_cmd.format('del', ptf_intf_ipv4_addr, intf_ipv4_addr.prefixlen, ptf_intf_name))
+    ptfhost.shell(ip_addr_config_cmd.format('del', ptf_intf_ipv6_addr, intf_ipv6_addr.prefixlen, ptf_intf_name))
+
+
+def test_arp_garp_enabled(intfs_for_test, config_facts, ptfadapter):
+    '''
+    Send a gratuitous ARP (GARP) packet from the PTF to the DUT
+
+    The DUT should learn the (previously unseen) ARP info from the packet
+    '''
+    arp_request_ip = '10.10.1.3'
+    arp_src_mac = '00:00:07:08:09:0a'
+    _, intf1_index, _, _, _, _, duthost = intfs_for_test
+
+    clear_dut_arp_cache(duthost)
+
+    vlan_intfs = config_facts['VLAN_INTERFACE'].keys()
+    garp_enable_cmd = 'redis-cli -n 4 HSET "VLAN_INTERFACE|{}" grat_arp enabled'
```

Comment:
These tests should be able to run on a single ToR topo (in addition to a dual ToR topo) so that we can verify these features without a hard requirement for a dual ToR setup.